### PR TITLE
Language organization

### PR DESCRIPTION
**Pull request by MestreWalla**
## Description
- Each language has its own file in the locales folder.

## Related Issues
- Issue #<158>

## Checklist
- [x] I have read and followed the [Contributing Guidelines](../CONTRIBUTING.md).
- [x] I have added necessary tests.
- [x] The project builds and runs without issues.

### CODE REVIEW

**Comment by MestreWalla** [comment]
-The reading order of the script files cannot be changed as it may break the code.
-I was unable to load the files dynamically as an array (the page always crashed when trying other methods as the other scripts did not recognize them).
-I could not import the scripts into another one as we are working on an offline project and this is not allowed.

**Comment by itz-rj-here** [comment]
Well. By using this method, in web store will show all the languages now.

**Comment by itz-rj-here** [comment]
@prem-k-r

**Comment by itz-rj-here** [comment]
#158

**Comment by itz-rj-here** [comment]
@MestreWalla If you don't mind I have a question. What about the width now?
Will it make such difference without the width limit?

**Comment by MestreWalla** [comment]
> @MestreWallaSe você não se importa, eu tenho uma pergunta. E quanto à largura agora? Fará tanta diferença sem o limite de largura?

I didn't understand what you meant.

**Comment by itz-rj-here** [comment]
> > @MestreWallaSe você não se importa, eu tenho uma pergunta. E quanto à largura agora? Fará tanta diferença sem o limite de largura?
> 
> I didn't understand what you meant.

image
These. The menu width and the other languages numberings...

**Comment by MestreWalla** [comment]
is still working inside languages.js. just the strings have been separated from the file

**Comment by itz-rj-here** [comment]
Uhh 😅
Idk actually. Wait @prem-k-r can you confirm this? If everything is good?..

**Comment by prem-k-r** [comment]
> Uhh 😅 Idk actually. Wait @prem-k-r can you confirm this? If everything is good?..

Seems good.. should be working fine.. if we want to include those menu width, numerals, comma, data . Etc.. it will be complicated.. we can keep it simple 

We can change file name to full code for some language pt-BR, es and zh.. not everywhere.. just file name

**Comment by prem-k-r** [comment]
There are some issue ike { id: 'googleEngineDD', key: 'googleEngine' }, we have redefine them now

**Comment by MestreWalla** [comment]
### 22/12/2024 - #01
- Refactor translation logic for improved maintainability.
These changes enhance code clarity, reduce redundancy, and facilitate future modifications.
- Fixed import font on css

**Comment by XengShi** [comment]
Conflicts!

**Comment by MestreWalla** [comment]
Conflicts reviewed, and tested.

> Conflicts!

File: languages.js [hunk 1]
```diff
@@ -2019,104 +73,123 @@ function localizeNumbers(text, language) {
...
+        'ai_tools',
+        'humidityLevel',
+        'feelsLike',
+        'location',
+        'googleEngine',
+        'duckEngine',
+        'bingEngine',
+        'braveEngine',
+        'youtubeEngine',
+        'googleEngineDD',
```
Comment (prem-k-r) [inline languages.js:137]: for 
{ id: 'todoListHeading', key: 'todoListText' },
{ id: 'googleEngineDD', key: 'googleEngine' },
        { id: 'duckEngineDD', key: 'duckEngine' },
        { id: 'bingEngineDD', key: 'bingEngine' },
        { id: 'braveEngineDD', key: 'braveEngine' },
        { id: 'youtubeEngineDD', key: 'youtubeEngine' },
        
id not equals key

File: locales/en.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    // Dialog boxes (alerts)
+    "confirmWallpaper": "Would you like to set a new image as your wallpaper for the day?",
+    "confirmRestore": "Are you sure you want to reset your settings? This action cannot be undone.",
+    "Nobackgroundset": "No background image is currently set.",
+    "clearbackgroundimage": "Are you sure you want to clear the background image?",
+    "confirmbackup": "Are you sure you want to backup your settings?",
+    "ProxyDisclaimer": "All proxy features are off by default.\n\nIf you enable search suggestions and CORS bypass proxy, it is strongly recommended to host your own proxy for enhanced privacy.\n\nBy default, the proxy will be set to https://mynt-proxy.rhythmcorehq.com, meaning all your data will go through this service, which may pose privacy concerns.",
+    "endlink": "There shouldn't be / at the end of the link",
+    "onlylinks": "Only links (starting with http:// or https://) are allowed.",
+    "imagedimensions": "Warning: The uploaded image dimensions ({width}x{height}) exceed 1920x1080 pixels. \n\nThis may impact performance or image may fail to load properly.",
```
Comment (prem-k-r) [inline locales/en.js:122]: todoListHeading, confirmbackup, imagedimensions
have been deleted in latest version  from all languages

File: .DS_Store
Comment (prem-k-r) [inline .DS_Store:1]: btw, is this .DS_Store file needed?

Comment (XengShi) [reply .DS_Store:1]: Not needed

Comment (prem-k-r) [reply .DS_Store:1]: there is also one at root folder, is that necessary for mac?

**Comment by XengShi** [comment]
Chinese language is not working!

**Comment by prem-k-r** [comment]
language code is incorrect, zh instead of zn

**Comment by prem-k-r** [comment]
I haven't checked on pc
image
are these transliteration showing?

File: locales/en.js (hunk 2 above)
Comment (MestreWalla) [reply locales/en.js:122]: Don't need these keys anymore?

File: languages.js (hunk 1 above)
Comment (MestreWalla) [reply languages.js:137]: resolved, not committed yet.

File: .DS_Store
Comment (MestreWalla) [reply .DS_Store:1]: nao possuo esse arquivo no projeto, funcionando normalmente.

File: locales/en.js (hunk 2 above)
Comment (prem-k-r) [reply locales/en.js:122]: Nope

Comment (prem-k-r) [reply locales/en.js:121]: also there are new additions
please refer https://github.com/XengShi/materialYouNewTab/discussions/195#discussioncomment-11650679

Comment (prem-k-r) [reply locales/en.js:121]: I can updates the languages if you want.. can raise PR in your branch

**Comment by MestreWalla** [comment]
Introduces a language analysis tool in the `tools` directory. 
This tool identifies complete and incomplete translations, displaying missing keys for incomplete languages.
Access is manual via the `tools` directory.

**Comment by itz-rj-here** [comment]
@prem-k-r

**Comment by XengShi** [comment]
@prem-k-r Please tell if this is ready for merge

**Comment by XengShi** [comment]
image

After changing the language it get stuck to `Feels` and `Earth` . previously it was reloading the page after changing the language. now it doesn't!

**Comment by itz-rj-here** [comment]
> image
> 
> After changing the language it get stuck to `Feels` and `Earth` . previously it was reloading the page after changing the language. now it doesn't!



> Owner

lemme test it out first
give me some time

**Comment by itz-rj-here** [comment]
image
Indonesian language looks weird here

**Comment by prem-k-r** [comment]
checked on pc, apart from missing translations and issue xengshi mentioned with weather.. rest are fine
(html yet to test)

**Comment by MestreWalla** [comment]
I'll try to resolve the issue without having to refresh the entire page, as I managed to do it badly. If I can't, I'll reinclude the update.

**Comment by prem-k-r** [comment]
wow, language analysis.. so cool

**Comment by prem-k-r** [comment]
@MestreWalla  there are some strings missing in some language, can I pull PR in you fork?

**Comment by MestreWalla** [comment]
Yes

**Comment by MestreWalla** [comment]
I returned the command to refresh the page;
I copied all the translations available in the ChengShi repository and replaced the ones in mine.
For me, the translations are correctly translated throughout the page in the pt language, but many languages ​​are incomplete. Using the analysis tool I created, translators can be helped in their languages.
Captura de Tela 2024-12-24 às 13 35 35
Captura de Tela 2024-12-24 às 13 35 53
Captura de Tela 2024-12-24 às 13 36 31

**Comment by prem-k-r** [comment]
>> @MestreWalla there are some strings missing in some language, can I pull PR in you fork?

> Yes

Opened  https://github.com/MestreWalla/materialYouNewTab/pull/1

**Comment by prem-k-r** [comment]
127 0 0 1_5500_languagesAnalysis html

**Review by prem-k-r — APPROVED** [review]

**Comment by prem-k-r** [comment]
@XengShi please review and merge this..

**Review by itz-rj-here — APPROVED** [review]

**Pull request merged**